tracking/src/DataCollection.js

```diff
@@ -0,0 +1,86 @@
+const NOT_AVAILABLE = "not available"
+class DataCollection {
+    constructor() {
+        console.log("DataCollection init"); 
+
+        this.getPlatform(); 
+        this.getLanguage(); 
+        this.getOS(); 
+        this.getScreenResolution(); 
+        this.getAppName(); 
+    }
+
+    toJson() {
+        const res = {}; 
+        res.p = this.platform; 
+        res.os = this.os; 
+        res.lang = this.language; 
+        res.sr = this.resolution; 
+        res.an = this.browser; 
+        return res; 
+    }
+
+    getPlatform() {
+        if (navigator) {
+            this.platform = navigator.platform || NOT_AVAILABLE; 
```

Comment:
if not available, then no need to set. Let the server try to guess from User Agent.

Reply:
changed